src/caselawclient/xquery/xslt_transform.xqy

```diff
@@ -14,6 +17,26 @@ let $xsl_path := fn:concat("judgments/xslts/", $xsl_filename)
 
 let $params := map:map()
 
+let $number_marks_xslt := (
+  <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                  version="2.0">
+    <xsl:template match="@*|node()">
+      <xsl:copy>
+        <xsl:apply-templates select="@*|node()"/>
+      </xsl:copy>
+    </xsl:template>
+    <xsl:template match="mark">
+      <xsl:copy>
+          <xsl:copy-of select="@*" />
+          <xsl:attribute name="id">
+              <xsl:text>mark_</xsl:text>
+              <xsl:value-of select="count(preceding::mark)"/>
```

Comment:
There's a part of me that's worried that this might be accidently quadratic -- if there's a thousand marks, does this take a half-million searches?

it's probably fine?

Reply:
As discussed on slack, it doesn't appear to be O(n^2) and is actually fairly performant on big judgments - we have however added a check to guard against gateway timeouts and return the unhighlighted judgment, logging a warning, so we can monitor.